service/src/main/java/bio/terra/workspace/common/utils/FlightUtils.java

```diff
@@ -111,4 +114,20 @@ public static String getFlightErrorMessage(FlightState subflightState) {
     }
     return errorMessage;
   }
+
+  /**
+   * Handles StorageException, when it has code 400 or 404, returns a STEP_RESULT_FAILURE_RETRY
+   * status to retry the step, otherwise, re-throw the exception.
+   *
+   * @param e storageException
+   * @return a stepResult with a retry status when the storageException has recoverable exception.
+   * @throws StorageException rethrows StorageException when the step should not be retried.
+   */
+  public static StepResult retryStorageExceptionOrRethrow(StorageException e) {
+    if (e.getCode() == HttpStatus.BAD_REQUEST.value()
```

Comment:
Do we need to retry both of these cases? I'm not actually sure what error code GCP returns if a group does not exist, you might have to check the integration test logs to verify